src/core/Table/utils.ts

```diff
@@ -29,3 +30,82 @@ export const getCellStyle = <T extends Record<string, unknown>>(
   }
   return style;
 };
+
+const isLastLeftStickyColumn = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  const index = columns.findIndex((x) => x.id === id);
+  const nextColumn = columns[index + 1];
+  return !nextColumn.sticky;
+};
+
+const getLeftMargin = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  let leftMargin = 0;
+  const index = columns.findIndex((x) => x.id === id);
+  for (let i = 0; i < index; i++) {
+    leftMargin += +columns[i].width! || 0;
+  }
+  return leftMargin;
+};
+
+const getRightMargin = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  let rightMargin = 0;
+  const index = columns.findIndex((x) => x.id === id);
+  for (let i = index + 1; i < columns.length; i++) {
+    rightMargin += +columns[i].width!;
+  }
+  return rightMargin;
+};
+
+const isLastRightStickyColumn = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  return columns[columns.length - 1].id === id;
+};
+
+export const getStickyCellStyle = <T extends Record<string, unknown>>(
+  tableInstance: TableInstance<T>,
+  column: ColumnInstance<T>,
+  isSelectable: boolean,
+): React.CSSProperties => {
+  const style = {} as React.CSSProperties;
+  if (column.id === SELECTION_CELL_ID) {
+    const isStickyTable = !!tableInstance.flatHeaders.find((x) => x.sticky);
+    if (isStickyTable) {
+      style.backgroundColor = '#FFF';
+      style.position = 'sticky';
+      style.zIndex = 1;
+      style.left = 0;
+      style.boxShadow = '-1px 0px 5px -2px gray';
+    }
+  }
+
+  if (column.sticky) {
+    style.backgroundColor = '#FFF';
```

Comment:
By default `style.backgroundColor` from 'iui-table-body' applies to rowgroup, not to the cell itself -> that's what makes non-sticky cells/columns overlap . Not sure how to avoid hardcoding it.